synapse/replication/tcp/redis.py

```diff
@@ -42,6 +43,24 @@
 
 logger = logging.getLogger(__name__)
 
+T = TypeVar("T")
+V = TypeVar("V")
+
+
+@attr.s(slots=True, frozen=True)
+class ConstantProperty(Generic[T, V]):
+    """A descriptor that returns the given constant, ignoring attempts to set
+    it.
+    """
+
+    constant = attr.ib()  # type: V
```

Comment:
FWIW we can now do the following I think?

```suggestion
    constant: V = attr.ib()
```

Reply:
I thought that was py3.6?

Reply:
Oh, maybe. I'm guessing this is why you need the `cast()` call?

Reply:
I believe it *should* understand descriptors enough to understand, but empirically it didn't. Possibly because of the `#type: ignore` (as mypy doesn't like overriding a bool attribute with a descriptor that returns a bool...)